clouddrift/wavelet.py

```diff
@@ -0,0 +1,808 @@
+       reach their maximum amplitudes. radian_frequency is between 0 and 2 * np.pi * 0.5,
+       the normalized Nyquist radian frequency.
+    order: int, optional
+        Order of wavelets, default is 1.
+    normalization: str, optional
+       Normalization for the ``wavelet`` output. By default it is assumed to be ``"bandpass"``
+       which uses a bandpass normalization, meaning that the FFT of the wavelets
+       have peak value of 2 for all central frequencies ``radian_frequency``. The other option is
+       ``"energy"``which uses the unit energy normalization. In this last case, the time-domain wavelet
+       energies ``np.sum(np.abs(wave)**2)`` are always unity.
+
+    Returns
+    -------
+    wavelet : np.ndarray
+        Time-domain wavelets. ``wavelet`` will be of shape (length,np.size(radian_frequency),order).
+    wavelet_fft: np.ndarray
+        Frequency-domain wavelets. ``wavelet_fft`` will be of shape (length,np.size(radian_frequency),order).
+
+    Examples
+    --------
+
+    Compute a Morse wavelet with gamma parameter 3, beta parameter 4, at radian frequency 0.2 cycles per unit time:
+
+    >>> wavelet, wavelet_fft = morse_wavelet(1024, 3, 4, np.array([2*np.pi*0.2]))
+    >>> np.shape(wavelet)
+    (1, 1, 1024)
+
+    Compute a suite of Morse wavelets with gamma parameter 3, beta parameter 4, up to order 3,
+    at radian frequencies 0.2 and 0.3 cycles per unit time:
+
+    >>> wavelet, wavelet_fft = morse_wavelet(1024, 3, 4, np.array([2*np.pi*0.2, 2*np.pi*0.3]), order=3)
+    >>> np.shape(wavelet)
+    (3, 3, 1024)
+
+    Compute a Morse wavelet specifying an energy normalization :
+    >>> wavelet, wavelet_fft = morse_wavelet(1024, 3, 4, np.array([2*np.pi*0.2]), normalization=energy)
+
+    Raises
+    ------
+    ValueError
+        If normalization optional argument is not in ["bandpass", "energy"]``.
+
+    See Also
+    --------
+    :func:`wavelet_transform`, `morse_wavelet_transform`, `morse_freq`, `morse_logspace_freq`, `morse_amplitude`, `morse_properties`
+    """
+    # ad test for radian_frequency being a numpy array
+    # initialization
+    wavelet = np.zeros((length, order, len(radian_frequency)), dtype=np.cdouble)
+    waveletfft = np.zeros((length, order, len(radian_frequency)), dtype=np.cdouble)
+
+    # call to morse_wavelet take only gamma and be as float, no array
+    fo, _, _ = morse_freq(gamma, beta)
+    for i in range(len(radian_frequency)):
+        wavelet_tmp = np.zeros((length, order), dtype=np.cdouble)
+        waveletfft_tmp = np.zeros((length, order), dtype=np.cdouble)
+
+        # wavelet frequencies
+        fact = np.abs(radian_frequency[i]) / fo
+        # norm_radian_frequency first dim is n points
+        norm_radian_frequency = (
+            2 * np.pi * np.linspace(0, 1 - 1 / length, length) / fact
+        )
+        if normalization == "energy":
+            waveletzero = np.exp(
+                beta * np.log(norm_radian_frequency) - norm_radian_frequency**gamma
```

Comment:
This always raises a RuntimeWarning because `norm_radian_frequency[0]` is always 0. Not the best UX. Should we assign the first element on a separate line?